### PR TITLE
Match some ROM cache functions

### PR DESCRIPTION
**Pull request by cadmic**
I added some crude documentation for some non-obvious struct fields too

### CODE REVIEW

File: src/rom.c [hunk 1]
```diff
@@ -852,10 +891,177 @@ s32 __romLoadBlock_Complete(Rom* pROM) {
...
+            return 0;
+        }
+
+        nOffsetRAM = iCacheRAM * 0x2000;
+        nOffsetARAM = iCacheARAM * 0x2000;
+        nOffsetARAM += ARGetBaseAddress();
+
+        while (ARGetDMAStatus()) {}
+
+        ARStartDMA(1, (u32)&pROM->pCacheRAM[nOffsetRAM], nOffsetARAM, 0x2000);
```
Comment (Yanis42) [inline src/rom.c:918]: I think the 1 in this function should be ``ARAM_DIR_ARAM_TO_MRAM``

File: src/rom.c [hunk 2]
```diff
@@ -852,10 +891,177 @@ s32 __romLoadBlock_Complete(Rom* pROM) {
...
+        nOffsetRAM = iCacheRAM * 0x2000;
+        nOffsetARAM = iCacheARAM * 0x2000;
+        nOffsetARAM += ARGetBaseAddress();
+
+        DCStoreRange(&pROM->pCacheRAM[nOffsetRAM], 0x2000);
+
+        while (ARGetDMAStatus()) {}
 
-#pragma GLOBAL_ASM("asm/non_matchings/rom/romMakeFreeCache.s")
+        ARStartDMA(0, (u32)&pROM->pCacheRAM[nOffsetRAM], nOffsetARAM, 0x2000);
```
Comment (Yanis42) [inline src/rom.c:939]: same here except it's ``ARAM_DIR_MRAM_TO_ARAM`` (according to [this](https://github.com/doldecomp/dolsdk2001/blob/e973c42aeb202d843e20006b94f408ce91e367c9/include/dolphin/ar.h#L6) at least)

this one's not defined btw

**Review by Yanis42 — COMMENTED** [review]
tiny details, otherwise good to go

File: src/rom.c (hunk 2 above)
Comment (Yanis42) [reply src/rom.c:939]: oohh 0 is "dma write" and 1 "dma read", should be worth adding a comment about it somewhere

Comment (Yanis42) [reply src/rom.c:939]: now that I think about it this call was most likely from the macro from ar.h

Comment (cadmic) [reply src/rom.c]: Thanks, I've replaced these with ARStartDMARead/ARStartDMAWrite

**Review by Yanis42 — APPROVED** [review]

**Pull request merged**